src/lat/push-lattice-test.cc

```diff
@@ -45,7 +45,8 @@ void TestPushCompactLatticeStrings() {
     if (s == 0)
       continue; // We don't check state zero, as the "leftover string" stays
                // there.
-    int32 first_label;
+    // Double check: HMM state ID's cannot be negative.
+    int32 first_label = -1;
```

Comment:
@danpovey I'm guessing you're fairly familiar with the CompactLattice code. Are the integers in the vector of integers in the Compact Lattice Weight required to be positive? I'm setting this to -1 as an "invalid value", since gcc isn't smart enough to figure out that first_label is guaranteed to be set before use in this code. Not that this initial value matters for this code.

Reply:
yes, they would always be positive.